flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/optimize/program/FlinkStreamProgram.scala

```diff
@@ -205,4 +217,17 @@ object FlinkStreamProgram {
 
     chainedProgram
   }
+
+  object JoinReorderMode extends Enumeration {
```

Comment:
duplicate with `FlinkBatchProgram#JoinReorderMode`, unify them into one and move it to a common package